src/main/java/bob/Bob.java

```diff
@@ -0,0 +1,47 @@
+package bob;
+
+import bob.command.Command;
+import bob.exception.BobException;
+
+public class Bob {
+    private final Storage storage;
+    private TaskList tasks;
+    private final Ui ui;
+
+    public Bob(String dataPath) {
+        ui = new Ui();
+        storage = new Storage();
+        try {
+            tasks = new TaskList(storage.load(dataPath));
+        } catch (BobException e) {
+            ui.showLoadingError(e.getMessage());
+            tasks = new TaskList();
+        }
+    }
+
+    private void runCommandLoopUntilExitCommand() {
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine();
+                Command command = Parser.parse(fullCommand);
+                command.execute(ui, storage, tasks);
+                isExit = command.isExit();
+            } catch (BobException e) {
+                ui.show(e.getMessage());
```

Comment:
great usage of try and catch block to handle your errors

Reply:
sflr thanks!